inspec/a2-iam-v2-1-only-integration/controls/role_projects.rb

```diff
@@ -0,0 +1,134 @@
+# encoding: utf-8
+# copyright: 2019, Chef Software, Inc.
+# license: All rights reserved
+
+title 'IAM v2.1 project filtering API integration tests'
+
+control 'iam-v2-projects-1' do
+  title 'v2.1-only access for project filtering'
+
+  CUSTOM_PROJECT_ID_1 = 'inspec-custom-project-1'
+  CUSTOM_PROJECT_ID_2 = 'inspec-custom-project-2'
+  CUSTOM_ROLE_ID_1 = 'inspec-custom-role-1'
+  CUSTOM_ROLE_ID_2 = 'inspec-custom-role-2'
+  CUSTOM_ROLE_ID_3 = 'inspec-custom-role-3'
+
+  CUSTOM_ROLE_1 = {
+          id: CUSTOM_ROLE_ID_1,
+          name: "Test Role 1",
+          actions: ["test:some:action", "test:other:action"],
+          projects: [CUSTOM_PROJECT_ID_1, CUSTOM_PROJECT_ID_2],
+          type: "CUSTOM"
+  }
+  CUSTOM_ROLE_2 = {
+          id: CUSTOM_ROLE_ID_2,
+          name: "Test Role 2",
+          actions: ["test:other:action"],
+          projects: [CUSTOM_PROJECT_ID_2],
+          type: "CUSTOM"
+  }
+  CUSTOM_ROLE_3 = {
+          id: CUSTOM_ROLE_ID_3,
+          name: "Test Role 2",
+          actions: ["test:other:action"],
+          projects: [],
+          type: "CUSTOM"
+  }
+
+  Roles = [ CUSTOM_ROLE_1, CUSTOM_ROLE_2, CUSTOM_ROLE_3 ]
+
+  CUSTOM_PROJECT_1 = {
+          id: CUSTOM_PROJECT_ID_1,
+          name: "Test Project 1"
+  }
+  CUSTOM_PROJECT_2 = {
+          id: CUSTOM_PROJECT_ID_2,
+          name: "Test Project 2"
+  }
+
+  Projects = [ CUSTOM_PROJECT_1, CUSTOM_PROJECT_2 ]
```

Comment:
Have not seen the pattern in our integration tests before, but I am less fluent in ruby so I can be fearless in trying to cover new ground. The advantage of defining all the above is that the `before-all` and `after-all` below are quite clean and tidy.

Reply:
Moving these to constants is perfect as long as we don't expect the tests to ever change them.